### PR TITLE
Image blit overlapped memory

### PR DESCRIPTION
**Pull request by aitor-lunarg**

### CODE REVIEW

**Comment by aitor-lunarg** [comment]
The new added tests are a copy paste from the image copy tests found in #4044, but modified to do a blit instead of a copy.

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -4408,21 +4516,22 @@ void CoreChecks::PreCallRecordCmdBlitImage(VkCommandBuffer commandBuffer, VkImag
                                            const VkImageBlit *pRegions, VkFilter filter) {
     StateTracker::PreCallRecordCmdBlitImage(commandBuffer, srcImage, srcImageLayout, dstImage, dstImageLayout, regionCount,
                                             pRegions, filter);
-    RecordCmdBlitImage(commandBuffer, srcImage, srcImageLayout, dstImage, dstImageLayout, regionCount, pRegions, filter);
+    RecordCmdBlitImage(commandBuffer, srcImage, srcImageLayout, dstImage, dstImageLayout, regionCount, pRegions, filter,
+                       CMD_BLITIMAGE);
```
Comment (jeremyg-lunarg) [inline layers/buffer_validation.cpp:4520]: This is what I just suggested in your other PR. :)

File: layers/state_tracker.cpp [hunk 2]
```diff
@@ -1687,6 +1687,12 @@ void ValidationStateTracker::PostCallRecordQueueBindSparse(VkQueue queue, uint32
                 auto image_state = Get<IMAGE_STATE>(bind_info.pImageOpaqueBinds[j].image);
                 auto mem_state = Get<DEVICE_MEMORY_STATE>(sparse_binding.memory);
                 if (image_state) {
+                    // An Android sepcial image cannot get VkSubresourceLayout until the image binds a memory.
+                    // See: VUID-vkGetImageSubresourceLayout-image-01895
+                    if (!image_state->fragment_encoder) {
+                        image_state->fragment_encoder = std::unique_ptr<const subresource_adapter::ImageRangeEncoder>(
+                            new subresource_adapter::ImageRangeEncoder(*image_state));
```
Comment (jeremyg-lunarg) [inline layers/state_tracker.cpp:1694]: same typos and make_unique as in other PR

File: layers/state_tracker.cpp [hunk 3]
```diff
@@ -1701,6 +1707,12 @@ void ValidationStateTracker::PostCallRecordQueueBindSparse(VkQueue queue, uint32
                 auto image_state = Get<IMAGE_STATE>(bind_info.pImageBinds[j].image);
                 auto mem_state = Get<DEVICE_MEMORY_STATE>(sparse_binding.memory);
                 if (image_state) {
+                    // An Android sepcial image cannot get VkSubresourceLayout until the image binds a memory.
+                    // See: VUID-vkGetImageSubresourceLayout-image-01895
+                    if (!image_state->fragment_encoder) {
+                        image_state->fragment_encoder = std::unique_ptr<const subresource_adapter::ImageRangeEncoder>(
```
Comment (jeremyg-lunarg) [inline layers/state_tracker.cpp:1713]: same typos and make_unique as in other PR

**Review by ncesario-lunarg — APPROVED** [review]
LGTM pending @jeremyg-lunarg's approval!

**Review by jeremyg-lunarg — COMMENTED** [review]
code LGTM. But there was a performance regression from another part of the sparse validation that merged last week. Do we want this to go in before that regression is understood / fixed?

**Comment by aitor-lunarg** [comment]
Tries to tackle #3993. There were performance concerns and issues with ycbcr images and CTS that need to be tackled before this can be merged at any point. Closing since I won't have time to go over this in the near future. Can be used as a reference.

**Pull request closed without merging**